### PR TITLE
Add RunListItem struct

### PR DESCRIPTION
**Pull request by jasonwbarnett**
No idea if this makes sense to merge into this code base considering it's not used in the code base. I wanted to tackle onto what was already done so I could use this functionality in my other projects.

### CODE REVIEW

**Comment by MarkGibbons** [comment]
Hi Jason,
Good to run into you again. I'll take a look at this over the next week.  Is this PR intended to make taking apart the returned runlists easier?  Any other intended use, or some other use intended?

**Comment by jasonwbarnett** [comment]
Hi Mark! Definitely a pleasure to run into you on the internet :)

This is the start of building out some code that will be a reverse proxy between chef clients and chef server to dynamically inject recipes/roles (run-list items) into the node's run list in a dynamic fashion. So this will make it easier to append, prepend, remove, or update any run-list items returned by Chef Server and then passed back to the chef client.

Next steps will be to update the Node struct with new types and then write a custom `MarshalJSON` and `UnmarshalJSON` for the RunList struct.

**Comment by MarkGibbons** [comment]
Still thinking about it.

File: run_list_item.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
+package chef
+
+import (
+	"fmt"
+	"regexp"
+)
+
+const (
```
Comment (MarkGibbons) [inline run_list_item.go]: Why are these constants exported from the package?  They look like they are for internal use only.

File: run_list_item.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	QualifiedRecipe            string = `^recipe\[([^\]@]+)(@([0-9]+(\.[0-9]+){1,2}))?\]$`
+	QualifiedRole              string = `^role\[([^\]]+)\]$`
+	VersionedUnqualifiedRecipe string = `^([^@]+)(@([0-9]+(\.[0-9]+){1,2}))$`
+	FalseFriend                string = `[\[\]]`
+)
+
+var (
```
Comment (MarkGibbons) [inline run_list_item.go]: These regexs look like they are internal package items.

File: run_list_item.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	QualifiedRecipeRegexp            *regexp.Regexp = regexp.MustCompile(QualifiedRecipe)
+	QualifiedRoleRegexp              *regexp.Regexp = regexp.MustCompile(QualifiedRole)
+	VersionedUnqualifiedRecipeRegexp *regexp.Regexp = regexp.MustCompile(VersionedUnqualifiedRecipe)
+	FalseFriendRegexp                *regexp.Regexp = regexp.MustCompile(FalseFriend)
+)
+
+type RunListItem struct {
```
Comment (MarkGibbons) [inline run_list_item.go]: We should have godoc comments on the functions and exported structures.

File: run_list_item.go (hunk 1 above)
Comment (jasonwbarnett) [reply run_list_item.go]: Updated!

File: run_list_item.go (hunk 3 above)
Comment (jasonwbarnett) [reply run_list_item.go]: Updated!

**Comment by jasonwbarnett** [comment]
@MarkGibbons updated this pull. Please review and let me know if it needs anything else.

**Comment by MarkGibbons** [comment]
Thanks Jason,

Could you add go style documentation for the Structs and functions.

https://golang.org/src/go/doc/example.go

Something like this:
// FunctionName is cool because and we have it here for

**Comment by MarkGibbons** [comment]
I'm trying to figure out how we would get from a runlist to a runlistitem that we would feed into your functions.  That seems to be a part of the code we are missing.

**Pull request merged**